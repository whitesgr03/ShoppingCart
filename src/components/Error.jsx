import "../style/error.css";
import Icon from "@mdi/react";
import { mdiHanger } from "@mdi/js";

import { Link } from "react-router-dom";

const Error = () => {
	return (
		<div className="error">
			<Icon path={mdiHanger} size={7} />

			<div>
				<h1>Service temporarily unavailable.</h1>
				<p>
					Our apologies, there has been an error. Please come back
					later or return to the <Link to="/">Home</Link> page. If you
					have any questions, please visit the{" "}
					<Link to="/contact">Contact us</Link>.
				</p>
			</div>
		</div>
	);
};

export { Error as default };
