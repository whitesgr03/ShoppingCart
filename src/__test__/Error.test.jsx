import { render, screen } from "@testing-library/react";

import Error from "../components/Error";

describe("Error Component", () => {
	it("Should render message if message is provide", () => {
		const mockMessage = "error";

		render(<Error message={mockMessage} />);

		const element = screen.getByRole("heading", {
			leave: 1,
			name: mockMessage,
		});

		expect(element).toBeInTheDocument();
	});
});
