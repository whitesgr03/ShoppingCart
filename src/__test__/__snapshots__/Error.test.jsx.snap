// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Error Component Should return Error DOM 1`] = `
<div>
  <div
    class="error"
  >
    <svg
      role="presentation"
      style="width: 10.5rem; height: 10.5rem;"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 4A3.5 3.5 0 0 0 8.5 7.5H10.5A1.5 1.5 0 0 1 12 6A1.5 1.5 0 0 1 13.5 7.5A1.5 1.5 0 0 1 12 9C11.45 9 11 9.45 11 10V11.75L2.4 18.2A1 1 0 0 0 3 20H21A1 1 0 0 0 21.6 18.2L13 11.75V10.85A3.5 3.5 0 0 0 15.5 7.5A3.5 3.5 0 0 0 12 4M12 13.5L18 18H6Z"
        style="fill: currentColor;"
      />
    </svg>
    <div>
      <h1>
        Service temporarily unavailable.
      </h1>
      <p>
        Our apologies, there has been an error. Please come back later or return to the 
        <a
          href="/"
        >
          Home
        </a>
         page. If you have any questions, please visit the
         
        <a
          href="/"
        >
          Contact us
        </a>
        .
      </p>
    </div>
  </div>
</div>
`;
